pkg/render/common/authentication/tigera/key_validator_config/key_validator_config.go

```diff
@@ -0,0 +1,133 @@
+package tigerakvc
+
+import (
+	"fmt"
+	"strconv"
+
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/authentication"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	defaultGroupsClaim   = "groups"
+	defaultUsernameClaim = "email"
+)
+
+// KeyValidatorConfig implements the KeyValidatorConfig interface. It uses the OIDC implicit flow, and maps the
+// wellknown config / jwks keys to a config map.
+type KeyValidatorConfig struct {
+	wellKnownConfig *authentication.WellKnownConfig
+	jwks            string
+	issuerURL       string
+	clientID        string
+	usernameClaim   string
+	groupsClaim     string
+	usernamePrefix  string
+	groupsPrefix    string
+}
+
+func New(issuerURL string, clientID string, options ...Option) (authentication.KeyValidatorConfig, error) {
+	kvc := &KeyValidatorConfig{
+		issuerURL:     issuerURL,
+		clientID:      clientID,
+		usernameClaim: defaultUsernameClaim,
+		groupsClaim:   defaultGroupsClaim,
+	}
+
+	for _, option := range options {
+		option(kvc)
+	}
+
+	wellKnownConfig, err := authentication.NewWellKnownConfig(kvc.issuerURL)
+	if err != nil {
+		return nil, err
+	}
+
+	jwks, err := wellKnownConfig.GetJWKS()
+	if err != nil {
+		return nil, err
+	}
+
+	kvc.wellKnownConfig = wellKnownConfig
+	kvc.jwks = string(jwks)
+
+	return kvc, nil
+}
+
+func (kvc *KeyValidatorConfig) Issuer() string {
+	return kvc.issuerURL
+}
+
+func (kvc *KeyValidatorConfig) ClientID() string {
+	return kvc.clientID
+}
+
+func (kvc *KeyValidatorConfig) RequiredEnv(prefix string) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{Name: fmt.Sprintf("%sOIDC_AUTH_ENABLED", prefix), Value: strconv.FormatBool(true)},
+		{Name: fmt.Sprintf("%sOIDC_AUTH_ISSUER", prefix), Value: kvc.Issuer()},
+		{Name: fmt.Sprintf("%sOIDC_AUTH_URL", prefix), Value: kvc.Issuer()},
```

Comment:
I'll double check this but I'm hesitant to remove this since this is what dex has been using, it may be that they can be the same but their not always?

Reply:
It actually doesn't look like this is an env variable for any project (i.e. voltron, es-proxy...)....